source/Camera/Advanced/AllocateMemoryForPointCloudData/AllocateMemoryForPointCloudData.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+This example shows two methods to copy point cloud data from GPU memory to CPU memory, to be consumed by OpenCV.
+With the first method, the memory is allocated by user (OpenCV) based on the camera resolution before any capture
+occurs. With the second method, the memory is allocated by Zivid SDK based on the point cloud resolution, so
+post-capture.
+
+Note: This example uses experimental SDK features, which may be modified, moved, or deleted in the future without
+notice.
+*/
+
+#include <Zivid/Experimental/SettingsInfo.h>
+#include <Zivid/Zivid.h>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <iostream>
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Creating settings" << std::endl;
+        auto settings = Zivid::Settings{ Zivid::Settings::Acquisitions{ Zivid::Settings::Acquisition{} } };
+
+        std::cout << "Getting camera resolution" << std::endl;
+        const auto resolution = Zivid::Experimental::SettingsInfo::resolution(camera.info(), settings);
+
+        std::cout << "Camera resolution:" << std::endl;
+        std::cout << "Height: " << resolution.height() << std::endl;
+        std::cout << "Width: " << resolution.width() << std::endl;
+
+        // Copy selected data from GPU to system memory (User-allocated)
+
+        std::cout << "Allocating the necessary storage with OpenCV API based on resolution info before any capturing"
+                  << std::endl;
+        auto rgbaUserAllocated = cv::Mat(resolution.height(), resolution.width(), CV_8UC4);
+
+        std::cout << "Capturing frame" << std::endl;
+        auto frame = camera.capture(settings);
+        auto pointCloud = frame.pointCloud();
+
+        std::cout << "Copying data with Zivid API from the GPU into the memory location allocated by OpenCV"
+                  << std::endl;
+        pointCloud.copyData(reinterpret_cast<Zivid::ColorRGBA *>(rgbaUserAllocated.data));
+
+        std::cout << "Displaying image" << std::endl;
+        cv::Mat bgrUserAllocated;
+        cv::cvtColor(rgbaUserAllocated, bgrUserAllocated, cv::COLOR_RGBA2BGR);
+        cv::imshow("BGR image User Allocated", bgrUserAllocated);
+        cv::waitKey(0);
+
+        // Copy selected data from GPU to system memory (Zivid-allocated)
+
+        std::cout << "Capturing frame" << std::endl;
+        frame = camera.capture(settings);
+        pointCloud = frame.pointCloud();
+
+        std::cout << "Copying colors with Zivid API from GPU to CPU" << std::endl;
+        auto colors = pointCloud.copyColorsRGBA();
+
+        std::cout << "Casting the data pointer as a void*, since this is what the OpenCV matrix constructor requires."
```

Comment:
Nitpick: I don't think we need this `cout` when we also have the comment below. Nothing actually happens at runtime when you cast, so there's no reason to display something at runtime imo. But it's not important, you can fix it if you have time.